### PR TITLE
chore(deps-checker): reduce duplicate deps checker calls

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Detail: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/14924646

The isInstalled() / getXXXVersion are called multiple times when the deps is installed. Reduce the duplicate calls.

### CODE REVIEW

File: packages/fx-core/src/common/deps-checker/depsChecker.ts [hunk 1]
```diff
@@ -1,18 +1,27 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 import { DepsCheckerError } from "./depsError";
-import { Result } from "@microsoft/teamsfx-api";
 
 export interface DepsChecker {
-  isInstalled(): Promise<boolean>;
+  getInstallationInfo(): Promise<DependencyStatus>;
```
Comment (swatDong) [inline packages/fx-core/src/common/deps-checker/depsChecker.ts:6]: Any test case to ensure `getInstallationInfo` called only once?

**Review by meifans — DISMISSED** [review]

File: packages/fx-core/tests/common/deps-checker/cases/dotnet.it.ts [hunk 2]
```diff
@@ -46,15 +49,15 @@ describe("DotnetChecker E2E Test - first run", async () => {
...
-    assert.isFalse(depsInfo.isLinuxSupported, "Linux should not support .NET");
+    assert.isFalse(depsInfo.isInstalled, ".NET is not installed, but isInstalled() return true");
+    assert.isFalse(depsInfo.details.isLinuxSupported, "Linux should not support .NET");
 
-    const res = await dotnetChecker.resolve();
-    assert.isTrue(res.isOk() && res.value);
+    const spyChecker = sandbox.spy(dotnetChecker);
+    const res = await spyChecker.resolve();
+    assert.isTrue(res.isInstalled);
+    assert.isTrue(spyChecker.getInstallationInfo.calledTwice);
```
Comment (swatDong) [inline packages/fx-core/tests/common/deps-checker/cases/dotnet.it.ts:60]: Why called twice? Is this fix to reduce to only one call?

Comment (XiaofuHuang) [reply packages/fx-core/tests/common/deps-checker/cases/dotnet.it.ts:60]: If it is the first time installation, will check whether the installation is success.

Comment (swatDong) [reply packages/fx-core/tests/common/deps-checker/cases/dotnet.it.ts:60]: make sense

**Review by swatDong — APPROVED** [review]

**Pull request merged**